core/2d/CCFastTMXLayer.cpp

```diff
@@ -356,7 +356,16 @@ void FastTMXLayer::setupTiles()
                 }
 
                 int pos = static_cast<int>(newX + _layerSize.width * y);
-                int gid = _tiles[pos];
+                uint32_t gid = _tiles[pos];
+                uint32_t flags = 0;
+
+                flags |= gid & kTMXTileHorizontalFlag;
+                flags |= gid & kTMXTileVerticalFlag;
+                flags |= gid & kTMXTileDiagonalFlag;
+
+                gid &= ~kTMXTileHorizontalFlag;
+                gid &= ~kTMXTileVerticalFlag;
+                gid &= ~kTMXTileDiagonalFlag;
```

Comment:
This can also be replaced with:
`gid &= ~kTMXFlipedAll;`

Reply:
> or
> gid &= kTMXFlippedMask;

wait.. don't you need to AND something with the complement to clear bits?

Reply:
They are defined as follows:

```
typedef enum TMXTileFlags_
{
    kTMXTileHorizontalFlag = 0x80000000,
    kTMXTileVerticalFlag   = 0x40000000,
    kTMXTileDiagonalFlag   = 0x20000000,
    kTMXFlipedAll          = (kTMXTileHorizontalFlag | kTMXTileVerticalFlag | kTMXTileDiagonalFlag),
    kTMXFlippedMask        = ~(kTMXFlipedAll)
} TMXTileFlags;
```

Reply:
Oh sorry I didn't notice the `kTMXFlippedMask` I thought it was `kTMXFlippedAll` that's what confused me

Reply:
Either way is fine, just use whatever is makes more sense within the context of the code you're writing (ie. whatever is more readable/requires less effort to figure out what is going on for anyone reading the code).